cdec/installation-manager-core/src/main/java/com/codenvy/im/managers/helper/NodeManagerHelperCodenvy4Impl.java

```diff
@@ -288,6 +306,39 @@ public void validateLicense() throws IOException {
         }
     }
 
+    private boolean isCodenvy4Installed() throws IOException {
+        Optional<Version> version = createArtifact(CDECArtifact.NAME).getInstalledVersion();
```

Comment:
What bad in it? It checks if the version is 4.x and so we must use specific operation in this case.